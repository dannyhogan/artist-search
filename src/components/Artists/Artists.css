.Artists {
  display: flex;
  flex-direction: column;
  list-style: none;
  padding: 0;

  li {
    display: flex;
    align-items: center;
    width: 400px;
    background: rgba(255, 255, 255, 0.136);
    border-radius: 15px;
    border: 2px solid black;
    margin: 10px;
    transition: .5s;

    h1 {
      margin: 10px;
    }

    a, a:visited {
      text-decoration: underline;
      color: black;
    }
  }

  li:hover {
    background: rgba(255, 255, 255, 0.565);
    transition: .5s;
    transform: scale(1.05);
  }
}
