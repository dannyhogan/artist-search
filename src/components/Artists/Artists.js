import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Artists.css';

export default function Artists({ artists }) {

  const artistLinks = artists.map(artist => {
    return <li key={artist.id}> <h1>{artist.name}</h1> <Link to={`/artist/${artist.name}/${artist.id}`}>View Releases</Link> </li>;
  });

  return (
    <ul className={styles.Artists}>
      {artistLinks}
    </ul>
  );
}

Artists.propTypes = {
  artists: PropTypes.array
};
