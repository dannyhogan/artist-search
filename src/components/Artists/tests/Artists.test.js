import React from 'react';
import { shallow } from 'enzyme';
import Artists from '../Artists';

describe('Artists component', () => {

  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({ ok: true }));
  });

  it('renders Artists component', () => {
    const artists = [{ name: 'Drake', id: '123' }];
    const wrapper = shallow(<Artists artists={artists} />);
    expect(wrapper).toMatchSnapshot();
  });
});
