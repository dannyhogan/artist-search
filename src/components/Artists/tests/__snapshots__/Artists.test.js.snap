// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artists component renders Artists component 1`] = `
<ul
  className="Artists"
>
  <li
    key="123"
  >
     
    <h1>
      Drake
    </h1>
     
    <Link
      to="/artist/Drake/123"
    >
      View Releases
    </Link>
     
  </li>
</ul>
`;
